core/src/main/scala/org/apache/spark/deploy/worker/ui/WorkerPage.scala

```diff
@@ -112,7 +112,15 @@ private[ui] class WorkerPage(parent: WorkerWebUI) extends WebUIPage("") {
       <td>
         <ul class="unstyled">
           <li><strong>ID:</strong> {executor.appId}</li>
-          <li><strong>Name:</strong> {executor.appDesc.name}</li>
+          <li><strong>Name:</strong>
+          {
+            if ({executor.state == ExecutorState.RUNNING}) {
+              <a href={executor.appDesc.appUiUrl}> {executor.appDesc.name}</a>
```

Comment:
Hi, @cloud-fan , @jiangxb1987 , @guoxiaolongzte . 
For a spark shell, we can disable WebUI like the following. In that case, that will be empty always.
```bash
bin/spark-shell --conf spark.ui.enabled=false
```